Add docs for rewriting history to claim authorship (#1269)
grafana/writers-toolkit: docs/sources/write/tooling-and-workflows/index.md

GitHub has detailed, cross-platform instructions for resolving a merge conflict using Git on the command line.
For more information refer to [Resolving a merge conflict using the command line](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line).

### Rewrite history to claim authorship

Sometimes you need to rewrite Git history to correct authorship information on commits.
This is useful to claim authorship of commits made by Cursor web agents that have the `cursorbot` author.

{{< admonition type="caution" >}}
Don't claim authorship of community contributions, contributors must sign the CLA.
{{< /admonition >}}

To change the author of the most recent commit:

```bash
git commit --amend --author="Your Name <your.email@example.com>"
```

To change the author of multiple commits, use interactive `git rebase`.
Replace the number `3` in `HEAD~3` with the number of commits you want to modify:

```bash
git rebase -i HEAD~3

> ```

Author: ````suggestion
To change the author of commits in your branch, use interactive `git rebase`.
The following command assumes you've branched from `main`.
If you've branched from another branch, replace `origin/main` with the name of that original branch.
```bash
git rebase -i "$(git merge-base HEAD origin/main)"
```
````

In the interactive editor, change `pick` to `edit` for each commit you want to modify.
Save and close the editor.

> For each commit marked for editing, Git pauses and allows you to make changes:

Author: ```suggestion
Git opens your text editor with the list of commits you're modifying.

1. In the interactive editor, change `pick` to `edit` for each commit you want to change the author for.
1. Save and close the editor.

For each commit marked for editing, Git pauses and allows you to make changes:
```

### Rewrite history to claim authorship

Sometimes you need to rewrite Git history to correct authorship information on commits.
This is useful to claim authorship of commits made by Cursor web agents that have the `cursorbot` author.

{{< admonition type="caution" >}}
Don't claim authorship of community contributions, contributors must sign the CLA.
{{< /admonition >}}

To change the author of commits in your branch, use interactive `git rebase`.
The following command assumes you've branched from `main`.
If you've branched from another branch, replace `origin/main` with the name of that original branch.
```bash
git rebase -i "$(git merge-base HEAD origin/main)"
```

Git opens your text editor with the list of commits you're modifying.

1. In the interactive editor, change `pick` to `edit` for each commit you want to change the author for.
1. Save and close the editor.

For each commit marked for editing, Git pauses and allows you to make changes:

```bash
git commit --amend --reset-author
git rebase --continue
```

Repeat this process for each commit you're editing.

Finally, push your changes to GitHub.

> The rebase rewrites history so you need to force push the rewrite to GitHub.

Author: ```suggestion
The `git rebase` command rewrites history so you need to force push the new history to GitHub.
```

GitHub has detailed, cross-platform instructions for resolving a merge conflict using Git on the command line.
For more information refer to [Resolving a merge conflict using the command line](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line).

> ### Rewrite history to claim authorship

Reviewer: Do I have to do this? What if I don't? What is the impact?

Sometimes you need to rewrite Git history to correct authorship information on commits.
This is useful to claim authorship of commits made by Cursor web agents that have the `cursorbot` author.

{{< admonition type="caution" >}}
Don't claim authorship of community contributions, contributors must sign the CLA.
{{< /admonition >}}

To change the author of commits in your branch, use interactive `git rebase`.
The following command assumes you've branched from `main`.
If you've branched from another branch, replace `origin/main` with the name of that original branch.
```bash
git rebase -i "$(git merge-base HEAD origin/main)"
```

> Git opens your text editor with the list of commits you're modifying.

Reviewer 2: A screenshot could be useful here!

> 1. In the interactive editor, change `pick` to `edit` for each commit you want to change the author for.

Reviewer 2: Click "START REBASE".

1. Save and close the editor.

For each commit marked for editing, Git pauses and allows you to make changes:

```bash
git commit --amend --reset-author
git rebase --continue
```

> 

Reviewer 2: Close Commit message tab.
